src/histogram.rs

```diff
@@ -0,0 +1,118 @@
+/// Structure used to create a distribution of node
+/// centrality values for use in a displayed histogram
+
+#[derive(Default, Clone)]
+pub struct Histogram {
+    /// Minimum value of a factor.
+    pub min: f64,
+    /// Maximum value of a factor.
+    pub max: f64,
+    /// store values
+    pub values: Vec<f64>,
+}
+
+impl Histogram {
+    pub fn default() -> Self {
+        Self {
+            min: f64::MAX,
+            max: f64::MIN,
+            values: Vec::new(),
+        }
+    }
+
+    // Store a single value
+    pub fn add(&mut self, value: f64) {
+        self.values.push(value);
+    }
+
+    // Compute the histogram tables, returning that table along
+    // with the value of the maximum count in a single slot.
+    pub fn compute(&mut self, num_counts: usize) -> (Vec<usize>, usize) {
+        for value in self.values.iter() {
```

Comment:
Nice 👍 . MinMax in single iteration - probably quicker than calling:
`self.values.iter().max()` and `self.values.iter().min()`